app/data_model/questionnaire_store.py

```diff
@@ -49,9 +50,8 @@ def get_json(self):
         return json.dumps(self.get_data())
 
     def has_changed(self):
-        current_hash = hash(self.get_json())
-
-        return current_hash != self.initial_hash
+        data = self.get_json()
+        return hash(data) != self.hash_on_load or data != self.data_on_load
```

Comment:
I can test it but probably not worth the effort. I imagine it will be marginally quicker but as we check if it has_changed once per request it's probably not needed